src/transformers/trainer.py

```diff
@@ -844,6 +836,48 @@ def create_optimizer(self):
 
         return self.optimizer
 
+    @staticmethod
+    def get_optimizercls_and_params(args: TrainingArguments) -> Tuple[Any, Any]:
+        """
+        Returns the optimizer class and optimizer parameters based on the training arguments.
+
+        Args:
+            args (`transformers.training_args.TrainingArguments`):
+                The training arguments for the training session.
+
+        """
+        optimizer_kwargs = {"lr": args.learning_rate}
+        adam_kwargs = {
+            "betas": (args.adam_beta1, args.adam_beta2),
+            "eps": args.adam_epsilon,
+        }
+        # TODO the following code is a good candidate for PEP 622 once Python 3.10 becomes the
+        #  minimum required version. See, https://www.python.org/dev/peps/pep-0622/
```

Comment:
this is very very unlikely to happen any time soon (like years), so just as well let's not add it?
```suggestion
```